jbrotli/src/main/java/org/meteogroup/jbrotli/io/BrotliInputStream.java

```diff
@@ -99,7 +100,12 @@ private void readChunkFromInput() throws IOException {
     while (!isEndOfInputStream && (uncompressedBufferPosition == 0 || brotliStreamDeCompressor.needsMoreInput())) {
       final int inLength = inputStream.read(in);
       if (inLength > 0) {
-        uncompressedBufferPosition += brotliStreamDeCompressor.deCompress(in, 0, inLength, uncompressedBuffer, uncompressedBufferPosition, uncompressedBuffer.length - uncompressedBufferPosition);
+        int[] lengths = brotliStreamDeCompressor.deCompress(in, 0, inLength, uncompressedBuffer, uncompressedBufferPosition, uncompressedBuffer.length - uncompressedBufferPosition);
+        if (lengths[0] < inLength) {
+          // TODO: handle this case
+          throw new IOException("Compressor did not consume all input");
```

Comment:
I don't like the idea of throwing an Exception here.
This is not an exception from the code flow, but an expected behavior, just in rare cases ;-)
Any other ideas?